### DISCUSSION TITLE
How do I push a branch whose name differs from its remote counterpart?

### DISCUSSION THREAD

**Discussion by ribbanya** [jesseduffield/lazygit#2130] [Q&A] [answered]
In git you can set `push.default` to `upstream` and then `git push` (with no further arguments) will push to the current branch's upstream, regardless of its name.

In lazygit, `P` executes `git push \"remote\" \"remote-branch-name\"` which results in the error "src refspec remote-branch-name does not match any".

Is there a way to change the behavior of `P`? I could overwrite the binding with a custom command, but then I wouldn't get the force prompt or other `PushCmdObj` behavior.

**Comment by mark2185** [comment, marked as answer]
You can set the upstream with `u` on a branch and then push as usual.

**Reply by ribbanya** [reply]
My upstream is already set.

**Reply by mark2185** [reply]
Are you on the latest `master`? Newer than `v0.35`, that is.

**Reply by ribbanya** [reply]
Ah, it was fixed in `v0.35`. Thank you.
